src/Microsoft.HttpRepl.Tests/Commands/ListCommandTests.cs

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.HttpRepl.Commands;
+using Microsoft.HttpRepl.Fakes;
+using Microsoft.HttpRepl.Preferences;
+using Microsoft.Repl.Parsing;
+using Xunit;
+
+namespace Microsoft.HttpRepl.Tests.Commands
+{
+    public class ListCommandTests : CommandTestsBase
+    {
+        [Fact]
+        public async Task ExecuteAsync_NoBaseAddress_ShowsWarning()
+        {
+            ArrangeInputs(commandText: "ls",
+                          baseAddress: "http://lcoalhost/",
+                          path: "/",
+                          urlsWithResponse: null,
+                          out MockedShellState shellState,
+                          out HttpState httpState,
+                          out ICoreParseResult parseResult,
+                          out _,
+                          out IPreferences preferences);
+
+            httpState.BaseAddress = null;
+            httpState.SwaggerEndpoint = new Uri("http://localhost/swagger.json");
+            httpState.Structure = new DirectoryStructure(null);
+
+            ListCommand listCommand = new ListCommand(preferences);
+
+            await listCommand.ExecuteAsync(shellState, httpState, parseResult, CancellationToken.None);
+
+            Assert.Single(shellState.Output);
```

Comment:
Just something to consider: if there were multiple lines out output returned, the test would fail without telling you what they were.  I.e. you'd have to run it through the debugger to determine the failed behavior.

You could instead do something like:

```
string actualOutput = String.Join(Environment.Newline, shellState.Output);
Assert.Equal(Resources.Strings.ListCommand_Error_NoBaseAddress, actualOutput);
```

This will fail if there are more than one line (because of the Join), but will give you the full output with a single assert.

Reply:
Good call. Changed.